File: Cpp/1-bit and 2-bit Characters.cpp
// 4ms, 75.55%; 8.6MB, 100.00%
class Solution
{
public:
    bool isOneBitCharacter(const vector<int> &bits)
    {
        int size = bits.size(), i = 0;
        while (i < size) {
            if (i + 1 == size) return true;
            i += (bits[i] ? 2 : 1);
        }
        return false;
    }
};

File: Python3/1-bit and 2-bit Characters.py
# 56ms, 92.10%; 13.9MB, 20.00%
class Solution:
    def isOneBitCharacter(self, bits: list) -> bool:
        size, i = len(bits), 0
        while i < size:
            if i + 1 == size: return True
            i += 1 if bits[i] == 0 else 2
        return False
